### PR TITLE
Add missing integration tests, clean up dockerfiles, and remove Anaconda dependencies

### PR DESCRIPTION
**Pull request by dcrankshaw**
This is a catch all maintenance PR that does a few things:
+ Removes all Docker images on Anaconda. The biggest change to functionality here is the removal of the  module dependency analysis that we did to try to automatically track down all Anaconda and pip dependencies. This feature relied on some semi-private Anaconda APIs and unfortunately was fairly buggy as a result. Furthermore, the conda installs in the docker images broke frequently for a variety of reasons.
+ Cleans up docker images and dramatically reduces the size of many of them. We now use the [`debian:stretch-slim`](https://hub.docker.com/_/debian/) Docker image as our base image. For the Python-based images that were previously using an Anaconda image as the base image, this hugely reduces image sizes and speeds up Docker build times.
+ Adds missing integration tests for MXNet, Pytorch, and the PyTorch->ONNX->Caffe2 deployer.
+ Adds a dev docker image
+ Consolidates all container entry scripts into a single, parameterized script

### CODE REVIEW

**Comment by dcrankshaw** [comment]
@chester-leung Can you look into why the mxnet test is failing?

**Comment by chester-leung** [comment]
Just figured out the problem. I've committed (but not pushed) changes. Will
also need to update documentation a little bit. I'll push changes by this
week's meeting.

**Comment by dcrankshaw** [comment]
Jenkins test this please

**Comment by dcrankshaw** [comment]
Jenkins test this please

**Comment by dcrankshaw** [comment]
jenkins test this please

**Comment by dcrankshaw** [comment]
jenkins test this please

File: dockerfiles/Py2RPCDockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+RUN mkdir -p /model \
+      && apt-get update \
+      && apt-get install -y libzmq5 libzmq5-dev redis-server libsodium18 build-essential
+
+RUN pip install cloudpickle==0.5.* pyzmq==17.0.* prometheus_client==0.1.* \
+    pyyaml==3.12.* jsonschema==2.6.* redis==2.10.* psutil==5.4.* flask==0.12.2 \
+    numpy==1.14.*
+
+
+# RUN conda install -y pyzmq libsodium
```
Comment (Corey-Zumar) [inline dockerfiles/Py2RPCDockerfile]: Remove commented code

File: dockerfiles/MXNetContainerDockerfile [hunk 2]
```diff
@@ -1,20 +1,10 @@
...
 
-RUN echo deb http://ftp.de.debian.org/debian jessie-backports main >> /etc/apt/sources.list \
-      && apt-get update --fix-missing \
-      && apt-get install -yqq -t jessie-backports openjdk-8-jdk \
-      && conda config --set ssl_verify no \
-      && conda install -c anaconda cloudpickle=0.5.2 \
-      && conda install -y --file /lib/python_container_conda_deps.txt \
-      && pip install mxnet==1.0.0 \
-      && apt-get install -y graphviz \
-      && pip install graphviz
```
Comment (Corey-Zumar) [inline dockerfiles/MXNetContainerDockerfile:14]: Is this `graphviz` package no longer required?

File: integration-tests/deploy_mxnet_models.py [hunk 3]
```diff
@@ -131,7 +126,7 @@ def get_test_point():
             mxnet_model = mx.mod.Module(softmax)
             mxnet_model.fit(data_iter, num_epoch=0)
```
Comment (Corey-Zumar) [inline integration-tests/deploy_mxnet_models.py:127]: Nit: `data_iter` isn't defined anywhere in this example. Consider defining it in the same fashion as the MXNet test case: `data_iter = mx.io.CSVIter(data_csv=train_path, data_shape=(785, ), batch_size=1)`

File: dockerfiles/Caffe2OnnxDockerfile [hunk 4]
```diff
@@ -1,23 +1,25 @@
...
-COPY containers/python/caffe2_onnx_container.py containers/python/caffe2_onnx_container_entry.sh /container/
-COPY clipper_admin/ /lib/clipper_admin
-RUN pip install /lib/clipper_admin
+ENV LD_LIBRARY_PATH=/usr/local/lib/ 
 
-CMD ["/container/caffe2_onnx_container_entry.sh"]
+RUN cd ~ && python -c 'from caffe2.python import core'
+
+RUN pip install onnx
+RUN python -c "import onnx"
```
Comment (Corey-Zumar) [inline dockerfiles/Caffe2OnnxDockerfile]: Just confirming that this isn't a piece of test code that should be removed (same for line 16); for debugging purposes, it seems helpful to verify correct import functionality for critical packages when the docker container is being built.

**Comment by dcrankshaw** [comment]
jenkins test this please

**Comment by shaneknapp** [comment]
we're experiencing some network issues from campus to the outside world.  i just ran `ping github.com` from one of the ubuntu workers and we're getting ~30% packet loss.  this definitely explains the jobs timing out.

**Review by Corey-Zumar — APPROVED** [review]
LGTM - just a nit and a question

File: dockerfiles/MXNetContainerDockerfile (hunk 2 above)
Comment (dcrankshaw) [reply dockerfiles/MXNetContainerDockerfile:14]: Graphviz was never required. It just got spuriously added.

File: integration-tests/deploy_mxnet_models.py (hunk 3 above)
Comment (dcrankshaw) [reply integration-tests/deploy_mxnet_models.py:127]: Not sure what you're referring to here? `data_iter` is defined on line 100/95 of this file.

File: dockerfiles/Caffe2OnnxDockerfile (hunk 4 above)
Comment (dcrankshaw) [reply dockerfiles/Caffe2OnnxDockerfile]: Yeah that's intentional

File: integration-tests/deploy_mxnet_models.py (hunk 3 above)
Comment (Corey-Zumar) [reply integration-tests/deploy_mxnet_models.py:127]: My mistake. I was referring to the "example" subsections of the documentation for the mxnet container. These sections don't define data_iter

Comment (dcrankshaw) [reply integration-tests/deploy_mxnet_models.py:127]: Ahh I see. Yeah good call, I'll add that.

**Comment by dcrankshaw** [comment]
jenkins test this please

**Comment by shaneknapp** [comment]
test this please
